src/docs/arch-tiered-storage.dox

```diff
@@ -0,0 +1,440 @@
+/*! @arch_page arch-tiered-storage Tiered Storage
+
+\warning Tiered storage is considered an experimental part of the WiredTiger
+library, and is not yet ready for production workloads.
+
+@section ts_intro Introduction and Definitions
+
+Tiered storage allows B-Trees to be stored into multiple places, more
+recently updated blocks on local disk and less recently updated blocks in cloud storage.
+The term \a object refers to one of these written
+btree parts, whether the object resides on the local disk, as an
+<em>object file</em>, or in the cloud, as a <em>cloud object</em>.
+The mechanism to create new objects is a WT_SESSION::checkpoint API call
+with the \c flush_tier configuration set.
+For brevity, is often called a \c flush_tier call.
+
+When in use, a tiered btree, like a non-tiered btree, may have some
+recently used or modified data that resides in memory pages.
+This in-memory representation is the same between a tiered btree
+and a non-tiered btree, it is only how data is stored on disk and in
+cloud objects that makes these btrees different.
+
+Tiered storage is configured and enabled on the connection
+via ::wiredtiger_open configuration options
+(e.g. <code>tiered_storage=(name="s3_store"),bucket="..."</code> ).
+Once enabled on the connection, the configuration applies to all tables created.
+The configuration can be overridden on individual tables can be changed with
+configuration options for WT_SESSION::create .
+This allows the caller to specify a different storage provider or bucket name or
```

Comment:
"This "what" allows the caller..."? I think it should be "The related WT_SESSION::create configuration options allow...". That should be written out clearly. I am waffling about whether the create config options should be specified here. I don't think it is necessary, but it did cross my mind.

Reply:
Fixed.  As to the create config options - this is the arch guide, so we I don't think we need to go over detailed stuff already in the reference manual.  The stuff that is duplicated here is purposeful just to give a little context for the following discussion.